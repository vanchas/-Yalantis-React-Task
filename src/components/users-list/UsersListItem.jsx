import React from 'react'
import styles from './users.module.css'

export default function UsersListItem({ user, index }) {
  return (
    <li className={styles.list_item}>
      <i>{index}</i> <b>{user.firstName} {user.lastName}</b>
    </li>
  )
}
