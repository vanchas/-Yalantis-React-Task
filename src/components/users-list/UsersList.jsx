import React from 'react'
import UsersListItem from './UsersListItem'
import styles from './users.module.css'

export default function UsersList({ users }) {

  return (
    <>
      {users && users.length ?
        <ul className={styles.list}>
          {users.map((user, index) => (
            <UsersListItem user={user} key={user.id} index={index + 1} />
          ))}
        </ul>
        : users && !users.length
          ? <div className={styles.message}>Sorry, there is no one user</div>
          : null
      }
    </>
  )
}

