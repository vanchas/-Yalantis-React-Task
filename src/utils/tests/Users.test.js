import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import UsersListItem from '../../components/users-list/UsersListItem'
import UsersList from '../../components/users-list/UsersList'

describe('Link', () => {
  it('should render correctly', () => {
    const user = {
      firstName: 'Mark', lastName: 'Tven', id: '1u8a7stgda7sg'
    }
    const output = shallow(
      <UsersListItem user={user} index={0} />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const users = [
      { firstName: 'Mark', lastName: 'Tven', id: '18f7tsdfg8sdt' },
      { firstName: 'Ilon', lastName: 'Mask', id: '2gas7itd7ait7' }
    ]
    const output = shallow(
      <UsersList users={users} />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})