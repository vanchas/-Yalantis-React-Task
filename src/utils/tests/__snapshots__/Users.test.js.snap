// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<li
  className="list_item"
>
  <i>
    0
  </i>
   
  <b>
    Mark
     
    Tven
  </b>
</li>
`;

exports[`Link should render correctly 2`] = `
<Fragment>
  <ul
    className="list"
  >
    <UsersListItem
      index={1}
      key="18f7tsdfg8sdt"
      user={
        Object {
          "firstName": "Mark",
          "id": "18f7tsdfg8sdt",
          "lastName": "Tven",
        }
      }
    />
    <UsersListItem
      index={2}
      key="2gas7itd7ait7"
      user={
        Object {
          "firstName": "Ilon",
          "id": "2gas7itd7ait7",
          "lastName": "Mask",
        }
      }
    />
  </ul>
</Fragment>
`;
